File: app/src/main/java/io/github/skywalkerdarren/simpleaccounting/model/entity/BillStats.java
package io.github.skywalkerdarren.simpleaccounting.model.entity;

import androidx.room.Entity;

import java.math.BigDecimal;

public class BillStats {
    private BigDecimal income;
    private BigDecimal expense;
    private BigDecimal sum;

    public BigDecimal getIncome() {
        return income;
    }

    public void setIncome(BigDecimal income) {
        this.income = income;
        sum = income.add(expense.negate());
    }

    public BigDecimal getExpense() {
        return expense;
    }

    public void setExpense(BigDecimal expense) {
        this.expense = expense;
        sum = income.add(expense.negate());
    }

    public BigDecimal getSum() {
        return sum;
    }

    public void setSum(BigDecimal sum) {
        this.sum = sum;
    }
}

File: app/src/main/java/io/github/skywalkerdarren/simpleaccounting/model/entity/BillStats.kt
package io.github.skywalkerdarren.simpleaccounting.model.entity

import java.math.BigDecimal

data class BillStats @JvmOverloads constructor(
        var income: BigDecimal = BigDecimal.ZERO,
        var expense: BigDecimal = BigDecimal.ZERO
) {
    val sum: BigDecimal
        get() = income.add(expense.negate())
}